### PR TITLE
fix(schema): Ensure all types of nested data are securely dispatched

### LINKED ISSUE #3003: Virtual properties of nested fields do not use external resolver
### Steps to reproduce

Resolve a virtual property inside a virtual property. The property that is 2 levels deep will not use its external resolver.

Example of 1 level deep working on external request, this is the result from the `roomOwners` service:
```
{
    "id": 11,
    "roomId": "11",
    "userId": "26",
    "user": {
        "id": 26,
        "tenantId": "26",
        "email": "test@test.com",
        "auth0Id": null,
        "name": null,
        "avatar": null
    }
}
```

Example of 2 levels deep not working on external request, this is the result from the `rooms` service:
```
{
    "id": 11,
    "name": "testroom",
    "description": "Test room",
    "createdAt": "1674456349778",
    "updatedAt": "1674456349778",
    "creatorId": "26",
    "tenantId": "26",
    ...
    "owners": [
        {
            "id": 26,
            "tenantId": "26",
            "email": "test@test.com",
            "password": "$2a$10$qw1jal6sxx6jvZO8LTgR/udtgYthA4ShCIq2hUbOWpuzsdZkEjHbO",
            "auth0Id": null,
            "name": null,
            "avatar": null
        }
    ]
}
```

Note, the password field is visible on the `user` property nested inside the `owners` property in the `rooms` service.

The resolver in roomOwners is here:
```
user: virtual(async (roomOwner, context) => {
    return context.app.service('users').get(roomOwner.userId);
}),
```

The resolver for the rooms is here:
```
owners: virtual(async (room, context) => {
    const { data } = await context.app.service('roomOwners').find({
        query: {
            roomId: room.id
        }
    });

    return data?.map((owner) => owner.user);
}),
```
Simple repository that reproduces this can be found here: https://github.com/edumeet/edumeet-management-server

### Expected behavior

For the external resolvers to run on nested properties.

### Actual behavior

The resolvers do not run.

**Module versions** (especially the part that's not working):
`^5.0.0-pre.35`

**NodeJS version**:
`18.13.0`

### PR DESCRIPTION
**Pull request by daffl**
Closes #3003

### CODE REVIEW

File: packages/schema/test/fixture.ts [hunk 1]
```diff
@@ -121,14 +121,24 @@ export const messageResolver = resolve<Message, HookContext<Application>>({
...
+
+    return user as Message['user']
   }),
   userList: virtual(async (_message, context) => {
     const users = await context.app.service('users').find({
       paginate: false
     })
 
-    return users as any
+    return users.map((user) => user) as any
```
Comment (marshallswain) [inline packages/schema/test/fixture.ts:141]: What's going on here?  Seems this line is not needed.

**Review by marshallswain — APPROVED** [review]
Looks good after that one line in the test is fixed.

File: packages/schema/test/fixture.ts (hunk 1 above)
Comment (daffl) [reply packages/schema/test/fixture.ts:141]: It was part of the test. Basically the `users` array did have the dispatch information but not the individual user entries so you ended up getting the unsafe data if you e.g. grabbed a user from the array or mapped it like in this case.

Comment (marshallswain) [reply packages/schema/test/fixture.ts:141]: Oh I see.

**Comment by marshallswain** [comment]
:shipit:

**Pull request merged**